docs/specs/openapi.yaml

```diff
@@ -238,44 +650,361 @@ paths:
+                      }
+                    ],
+                    "ready": true,
+                    "spawn-time": "2024-12-27T12:31:26.673297951+08:00",
+                    "ready-time": "2024-12-27T12:31:27.686371869+08:00"
+                  }
+                }
+  /v1/notices:
+    get:
+      summary: Get notices
+      tags:
+        - notices
+      description: Get a list of notices that match the filters, ordered by the last-repeated time.
+      parameters:
+        - in: query
+          name: user-id
+          description: Filter notices by user ID (admin only). Only one user ID can be specified.
+          schema:
+            type: integer
+            format: int32
+        - in: query
+          name: users
+          description: If set to "all", return notices for all users (admin only). Cannot be used with `user-id`.
+          schema:
+            type: string
+            enum: ["all"]
+        - in: query
+          name: types
+          description: Filter notices by type. Multiple types can be specified as a comma-separated list.
+          schema:
+            type: array
+            items:
+              type: string
+              enum: [change-update, custom, warning]
+        - in: query
+          name: keys
+          description: Filter notices by keys. Multiple keys can be specified as a comma-separated list.
+          schema:
+            type: array
+            items:
+              type: string
+        - in: query
+          name: after
+          description: Filter notices occurring after the specified timestamp.
+          schema:
+            type: string
+            format: date-time
+        - in: query
+          name: timeout
+          description: The maximum time [duration](#duration) to wait for notices. If no notices are available within this time, an empty list is returned.
+          schema:
+            type: string
+            format: duration
+      responses:
+        "200":
+          description: Successful response.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetNoticesResponse"
+              example:
+                {
+                  "type": "sync",
+                  "status-code": 200,
+                  "status": "OK",
+                  "result": [
+                    {
+                      "id": "1",
+                      "user-id": null,
+                      "type": "change-update",
+                      "key": "1",
+                      "first-occurred": "2024-12-27T09:55:13.393868798Z",
+                      "last-occurred": "2024-12-27T09:55:14.400978382Z",
+                      "last-repeated": "2024-12-27T09:55:14.400978382Z",
+                      "occurrences": 3,
+                      "last-data": {
+                        "kind": "autostart"
+                      },
+                      "expire-after": "168h0m0s"
```

Comment:
Should we also include `retry-after` in this example (and the example for POST /v1/notices/{id})? I don't think it's essential, I just wondered whether it's not here for some reason.

Reply:
I tried both change-update notices and custom notices, and neither returned `repeat-after`, so I left them out and used real responses from Pebble.